openml-h2o/src/main/java/com/feedzai/openml/h2o/algos/H2OAlgoUtilsFactory.java

```diff
@@ -0,0 +1,78 @@
+package com.feedzai.openml.h2o.algos;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.feedzai.openml.data.schema.DatasetSchema;
+import com.feedzai.openml.h2o.H2OAlgorithm;
+import com.feedzai.openml.h2o.server.H2OApp;
+import com.feedzai.openml.provider.descriptor.MLAlgorithmDescriptor;
+import com.feedzai.openml.util.algorithm.MLAlgorithmEnum;
+
+/**
+ * Factory class responsible for providing the correct H20 Algorithm util.
+ *
+ * @author Antonio Silva (antonio.silva@feedzai.com)
+ * @since @@@feedzai.next.release@@@
+ */
+public class H2OAlgoUtilsFactory {
+
+    /**
+     * Logger for this class.
+     */
+    private static final Logger logger = LoggerFactory.getLogger(H2OApp.class);
+
+    /**
+     * Private constructor to avoid class instantiation.
+     *
+     * @since @@@feedzai.next.release@@@
+     */
+    private H2OAlgoUtilsFactory() {
+    }
+
+    /**
+     * Factory main method to get the correct H20 Algo Utils.
+     * @param algorithmDescriptor descriptor for the algorithm we want the utils for.
+     * @param schema              datasource schema.
+     * @return AbstractH2OAlgoUtils implementation for the algoDescriptor/schema provided.
+     *
+     * @since @@@feedzai.next.release@@@
+     */
+    public static AbstractH2OAlgoUtils getH2OAlgoUtils(final MLAlgorithmDescriptor algorithmDescriptor,
+                                                       final DatasetSchema schema) {
+        //TODO: the AbstractH2OAlgoUtils objects have no state, create one for each call or have singletons to return ?
```

Comment:
remove?

Reply:
Removed but I still think that in the future we could improve this so we don't create a new object every time